sensors/messaging_sensor.py

```diff
@@ -0,0 +1,92 @@
+from keystoneclient.v3 import client as keystone
+from zaqarclient.queues.v1 import client as zaqar
+
+from st2reactor.sensor.base import PollingSensor
+
+
+class MessageQueueSensor(PollingSensor):
+    """
+    * self.sensor_service
+        - provides utilities like
+            - get_logger() - returns logger instance specific to this sensor.
+            - dispatch() for dispatching triggers into the system.
+    * self._config
+        - contains parsed configuration that was specified as
+          config.yaml in the pack.
+    """
+
+    def setup(self):
+        super(MessageQueueSensor, self).setup()
+        self.logger = self.sensor_service.get_logger(name=self.__class__.__name__)
+
+    def _get_keystone_auth_cred(self):
+        return {
+            'username': self._config['password']['OS_USERNAME'],
+            'password': self._config['password']['OS_PASSWORD'],
+            'auth_url': self._config['password']['OS_AUTH_URL'],
+            'project_id': self._config['password']['OS_TENANT_ID']
+        }
+
+    def _get_zaqar_auth_config(self, auth_token):
+        return {
+            'auth_opts': {
+                'options': {
+                    'os_auth_token': auth_token,
+                    'os_project_id': self._config['password']['OS_TENANT_ID'],
+                    'os_auth_url': self._config['password']['OS_AUTH_URL']
+                }
+            }
+        }
+
+    def _get_claim_options(self):
+        return {
+            'ttl': self._config['messaging']['claim_ttl'],
+            'grace': self._config['messaging']['claim_grace']
+        }
+
+    def _dispatch_message(self, queue_name, message):
+        payload = {'queue': queue_name, 'message': message}
+        self.sensor_service.dispatch(trigger='openstack.messaging', payload=payload)
+
+    def poll(self):
+        keystone_cred = self._get_keystone_auth_cred()
+        keystone_cli = keystone.Client(**keystone_cred)
+
+        messaging_endpoint = keystone_cli.service_catalog.url_for(
+            service_type=self._config['messaging']['service_type'],
+            endpoint_type='publicURL'
+        )
+
+        messaging_cli = zaqar.Client(
+            messaging_endpoint,
+            conf=self._get_zaqar_auth_config(keystone_cli.auth_token),
+            version=2
+        )
+
+        claim_options = self._get_claim_options()
+        claim_ttl = claim_options['ttl']
+        claim_grace = claim_options['grace']
+
+        for queue_name in self._config['messaging']['queues']:
```

Comment:
Getting queues from global config makes one restart the sensor container when they change. 
Better way is to do it in trigger add / remove:
https://github.com/StackStorm/st2/blob/master/contrib/linux/sensors/file_watch_sensor.py#L35-L42